src/components/tables/EntityTable.tsx

```diff
@@ -10,51 +12,208 @@ import {
     TablePagination,
     TableRow,
     TableSortLabel,
+    TextField,
+    Toolbar,
     Tooltip,
+    Typography,
 } from '@mui/material';
+import { SupabaseClient } from '@supabase/supabase-js';
+import ExternalLink from 'components/shared/ExternalLink';
 import formatDistanceToNow from 'date-fns/formatDistanceToNow';
-import { MouseEvent, useState } from 'react';
+import { ChangeEvent, MouseEvent, useEffect, useState } from 'react';
 import { FormattedMessage, useIntl } from 'react-intl';
-import {
-    DeploymentStatus,
-    Entity,
-    EntityMetadata,
-} from 'stores/ChangeSetStore';
+import { TABLES } from 'services/supabase';
+import { PostgrestError, useClient } from 'supabase-swr';
+
+enum TableStatuses {
+    LOADING = 'LOADING',
+    DATA_FETCHED = 'DATA_FETCHED',
+    NO_EXISTING_DATA = 'NO_EXISTING_DATA',
+    TECHNICAL_DIFFICULTIES = 'TECHNICAL_DIFFICULTIES',
+    UNMATCHED_FILTER = 'UNMATCHED_FILTER',
+}
+
+type TableStatus =
+    | TableStatuses.LOADING
+    | TableStatuses.DATA_FETCHED
+    | TableStatuses.NO_EXISTING_DATA
+    | TableStatuses.TECHNICAL_DIFFICULTIES
+    | TableStatuses.UNMATCHED_FILTER;
+
+type DeploymentStatus = 'ACTIVE' | 'INACTIVE';
 
 type SortDirection = 'asc' | 'desc';
 
 interface Props {
-    entities: Entity[];
-    updateDeploymentStatus: (
-        key: string,
-        deploymentStatus: DeploymentStatus
-    ) => void;
+    noExistingDataContentIds: {
+        header: string;
+        message: string;
+        docLink: string;
+        docPath: string;
+    };
+}
+
+interface Entity {
```

Comment:
This interface was formally known as `EntityMetadata` and located in _src/stores/ChangeSetStore.ts_. The original `Entity` interface was broken into two parts: _metadata_ and _resources_. Since the UI is no longer tasked with maintaining and keeping record of the resources associated with a given entity, the concept of an _entity_ has been reduced to what was formally labelled _entity metadata_ (as far as this component is concerned).